src/Board.cpp

```diff
@@ -0,0 +1,192 @@
+#include "Board.h"
+#include "Piece.h"
+#include "Rook.h"
+#include "King.h"
+#include "Bishop.h"
+#include "Queen.h"
+#include "Pawn.h"
+#include "Knight.h"
+#include <memory>
+
+const int white_pawn_row = 1;
+const int black_pawn_row = 6;
+
+Board::Board()
+{
+	initializeBoard();
+	_black_king = _board[7][4];
+	_white_king = _board[0][4];
+}
+
+Board::~Board() {
+	for (int i = 0; i < ROW_LENGTH; i++)
+	{
+		for (int j = 0; j < COL_LENGTH; j++)
+		{
+			if (_board[i][j] != nullptr)
+			{
+				delete _board[i][j];
+				_board[i][j] = nullptr;
+			}
+		}
+	}
+	_white_king = nullptr;
+	_black_king = nullptr;
+}
+void Board::initializeBoard() {
+
+	for (int i = 0; i < ROW_LENGTH; i++)
+		for (int j = 0; j < COL_LENGTH; j++)
+			_board[i][j] = nullptr;
+
+	_current_player = WHITE;
```

Comment:
creating the pieces manually and not by the string passed to the chess class
if i change the string for testing i can play with empty pieces

Reply:
I do not understand the comment.